src/main/scala/mesosphere/marathon/storage/migration/Migration.scala

```diff
@@ -100,6 +100,13 @@ class Migration(
 
   @SuppressWarnings(Array("all")) // async/await
   def migrateAsync(): Future[Seq[StorageVersion]] = async {
+    // Before reading to and writing from the storage, let's ensure that
+    // no stale values are read from the persistence store.
+    // Although in case of ZK it is done at the time of creation of CuratorZK,
+    // it is better to be safe than sorry.
+    logger.info(s"$persistenceStore")
+    await(persistenceStore.sync())
```

Comment:
So what happens if `sync` fails here (e.g. `StoreCommandFailedException`)? Did you test this case?

Reply:
Yes. I will add a test.